### ISSUE TITLE
onstart is called after dataAvailable fired

### ISSUE THREAD

**Issue by rotem925** [chris-rudmin/opus-recorder#171]
Hi Chris,

I see a new issue that wasn't there on version 5.0
Seems that the dataAvailable callback is called before the onstart is fire.
I'm referencing this old issue again #132

**Comment by chris-rudmin** [comment]
@rotem925 Regression :( I'll give it a look.

**Comment by chris-rudmin** [comment]
@rotem925 I was unable to write a good unit test for my fix.  Could you test it on your side and validate?

**Comment by chris-rudmin** [comment]
@rotem925 I made a second attempt to fix this (new commits to the same branch).  Care to try it out?

**Comment by rotem925** [comment]
@chris-rudmin No success here also
Is my fix not enough?

**Comment by chris-rudmin** [comment]
@rotem925 I aprreciate your suggestion, but your fix doesn't make sense to me.  You are running `onstart` (which I assume returns a promise in your implementation) in parallel with initSourceNode and initWorker.  It is not "correct" in the sense that likely the recording has not started yet when it is called.

The last fix I made only generates the header pages after `onstart` is called, so I don't understand how the `onstart` callback could be called after pages are generated.

**Comment by rotem925** [comment]
Chris,
Can you try it on your side on the example page as well?
All I do is putting a log on the start event and on the dataAvailable event.

Rotem

**Comment by chris-rudmin** [comment]
@rotem925 I tested and validated my fix works.
